lib/hem/tasks/magento2/createpatch.rb

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env ruby
+# ^ Syntax hint
+
+desc 'Create magento2 patch'
+task :createpatch do
+  package_name = Hem.ui.ask("which package did you modify? (e.g. module-catalog)")
+  patch_name = Hem.ui.ask("what should be the name of the patch? (e.g. my awesome fix )")
+  patch_tmp_path = '/tmp/patchtmp/'
+  package_path = 'vendor/magento/' + package_name
+
+  if patch_name.empty?
+    Kernel.abort('patch name required')
+  end
+
+  if package_name.empty?
+    Kernel.abort('package name required')
+  end
+
+  if !File.exists?(package_path)
+    Kernel.abort(package_path + ' does not exist')
+  end
+
+  Hem.ui.info('Patch will be created for ' + package_path)
+  Hem.ui.info('1. Save your changes to ' + patch_tmp_path + package_name)
+  FileUtils.rm_rf(patch_tmp_path)
+  FileUtils.mkdir_p(patch_tmp_path)
+  FileUtils.cp_r(package_path, patch_tmp_path)
+
+  Hem.ui.info('2. Reinstall package')
+  FileUtils.rm_rf(package_path)
+  run_command('php bin/composer.phar install', realtime: true, indent: 2)
+
+  Hem.ui.info('3. Init git with the original code of the package')
+  FileUtils.cd(package_path)
+  shell('git init')
+  shell('git add .')
+  shell('git commit -m "original code"')
```

Comment:
Can you also use the `--no-gpg-sign` option here, since it's a temporary commit it shouldn't require involvement by those that use the `commit.autosign` option.